components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -259,6 +261,246 @@
         }
     }
 
+    /**
+     * Get all basic identity provider information for a matching filter.
+     *
+     * @param tenantId       tenant Id of the identity provider.
+     * @param expressionNode list of filter value for IdP search.
+     * @param limit          limit per page.
+     * @param offset         offset value.
+     * @param sortOrder      order of IdP ASC/DESC.
+     * @param sortBy         the attribute need to sort.
+     * @return Identity Provider's Basic Information list.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    List<IdentityProvider> getIdPsSearch(int tenantId, List<ExpressionNode> expressionNode, int limit, int offset,
```

Comment:
Too lengthy method and hard to understand. Refactor to make it easy to understand the logic.